politeiawww/proposals_test.go

```diff
@@ -457,49 +495,69 @@ func TestValidateProposal(t *testing.T) {
 				ErrorCode: www.ErrorStatusInvalidSignature,
 			}},
 
+		{"invalid public key", *propInvalidPubKey, usr,
+			www.UserError{
+				ErrorCode: www.ErrorStatusInvalidPublicKey,
+			}},
+
 		{"incorrect signature", *propBadSig, usr,
 			www.UserError{
 				ErrorCode: www.ErrorStatusInvalidSignature,
 			}},
 
+		{"no index file", *propNoIndexFile, usr,
+			www.UserError{
+				ErrorCode: www.ErrorStatusNoIndexFile,
+			}},
+
 		{"no files", *propNoFiles, usr,
 			www.UserError{
-				ErrorCode: www.ErrorStatusProposalMissingFiles,
+				ErrorCode: www.ErrorStatusNoIndexFile,
 			}},
 
-		{"bad md filename", *propBadFilename, usr,
+		{"invalid index filename", *propInvalidFilename, usr,
 			www.UserError{
-				ErrorCode: www.ErrorStatusProposalMissingFiles,
+				ErrorCode: www.ErrorStatusNoIndexFile,
 			}},
 
-		{"duplicate filenames", *propDupFiles, usr,
+		{"invalid index mime type", *propInvalidIndexMimeType, usr,
 			www.UserError{
-				ErrorCode: www.ErrorStatusProposalDuplicateFilenames,
+				ErrorCode: www.ErrorStatusInvalidMIMEType,
 			}},
 
-		{"too may md files", *propMaxMDFiles, usr,
+		{"invalid title", *propBadTitle, usr,
 			www.UserError{
-				ErrorCode: www.ErrorStatusMaxMDsExceededPolicy,
+				ErrorCode: www.ErrorStatusProposalInvalidTitle,
 			}},
 
-		{"too many images", *propMaxImages, usr,
+		{"too many attached files", *propMaxAttachments, usr,
 			www.UserError{
-				ErrorCode: www.ErrorStatusMaxImagesExceededPolicy,
+				ErrorCode: www.ErrorStatusMaxAttachmentsExceededPolicy,
 			}},
 
-		{"md file too large", *propMDLarge, usr,
+		{"attachment file too large", *propAttachmentLarge, usr,
 			www.UserError{
-				ErrorCode: www.ErrorStatusMaxMDSizeExceededPolicy,
+				ErrorCode: www.ErrorStatusMaxAttachmentSizeExceededPolicy,
 			}},
 
-		{"image too large", *propImageLarge, usr,
+		{"too many index files", *propMaxIndexFiles, usr,
 			www.UserError{
-				ErrorCode: www.ErrorStatusMaxImageSizeExceededPolicy,
+				ErrorCode: www.ErrorStatusMaxIndexFileExceededPolicy,
```

Comment:
Tests should be in the order in which the errors appear in the validateProposal function.  It makes them easier to maintain and to know when a test is missing.  For example the validateProposal tests should be in the order:

```
ErrorStatusInvalidSignature
ErrorStatusInvalidPublicKey
ErrorStatusNoIndexFile
ErrorStatusInvalidMIMEType
ErrorStatusNoIndexFile
ErrorStatusMaxIndexFileExceededPolicy
ErrorStatusMaxAttachmentsExceededPolicy
ErrorStatusMaxIndexFileSizeExceededPolicy
ErrorStatusMaxAttachmentSizeExceededPolicy
ErrorStatusProposalDuplicateFilenames
ErrorStatusProposalInvalidTitle
ErrorStatusInvalidSignature
success
```

If any of them were previously out of order, please fix them.

Reply:
Validation tests are in order now.